### ISSUE TITLE
Add support for HTTP Digest Authentication

### ISSUE THREAD

**Issue by kdckrs** [Smile-SA/elasticsuite#2163]
**Is your feature request related to a problem? Please describe.**
When trying to connect our Magento 2 Smile-SA Elastic Suite module with an Elasticsearch instance on AWS (with fine-grained-security enabled), we are unable to perform the authentication using Basic HTTP Authentication.

When applying the following settings in the configuration:

```
...
'smile_elasticsuite_core_base_settings' => [
    'es_client' => [
        ...
        'enable_http_auth' => 1,
        'http_auth_user' => <some-username>',
        'http_auth_pwd' => '<some-super-secret-password>'
    ],
    ...
]
...
```

it results in a request that is made using the HTTP Basic authentication "<some-username>:<some-super-secret-password>".

The AWS Elasticsearch instance only supports requests using the HTTP Digest Authentication where the credentials are passed via a hashed version of the username and password.

**Describe the solution you'd like**
Add support for Digest Authentication.

**Describe alternatives you've considered**
/

**Additional context**
- RFC 2617  https://www.ietf.org/rfc/rfc2617.txt

**Comment by romainruaud** [comment]
Hi @kdckrs 

I was not aware of this type of authentication for AWS Elasticsearch instances.

I'm not even sure if the elasticsearch php client is supporting it, we should check before trying something.

Do you feel comfortable with proposing a PR to us for this one ?

**Comment by kdckrs** [comment]
Hi @romainruaud ,

Yes, it you enable fine-grained security when creating a new Elasticsearch Instance it ships with Open Distro features: https://opendistro.github.io/for-elasticsearch/ , which includes Basic HTTP Authentication for internal users.

I have created a PR with the proposed changes so you can have a look ;) , see https://github.com/Smile-SA/elasticsuite/pull/2167

**Comment by romainruaud** [comment]
PR was merged and published into 2.10.5 version.
